index/secondary-index/src/test/scala/org/apache/carbondata/spark/testsuite/mergeindex/CarbonIndexFileMergeTestCaseWithSI.scala

```diff
@@ -215,52 +238,59 @@ class CarbonIndexFileMergeTestCaseWithSI
         s"'GLOBAL_SORT_PARTITIONS'='20')")
     sql("CREATE INDEX nonindexmerge_index4 on table nonindexmerge (name) AS 'carbondata'")
     val rows = sql("""Select count(*) from nonindexmerge""").collect()
-    assert(getIndexFileCount("default_nonindexmerge", "0") == 20)
-    assert(getIndexFileCount("default_nonindexmerge", "1") == 20)
-    assert(getIndexFileCount("default_nonindexmerge", "2") == 20)
-    assert(getIndexFileCount("default_nonindexmerge", "3") == 20)
-    assert(getIndexFileCount("default_nonindexmerge_index4", "0") == 20)
-    assert(getIndexFileCount("default_nonindexmerge_index4", "1") == 20)
-    assert(getIndexFileCount("default_nonindexmerge_index4", "2") == 20)
-    assert(getIndexFileCount("default_nonindexmerge_index4", "3") == 20)
+    assert(CarbonTestUtil.getIndexFileCount("default_nonindexmerge", "0",
```

Comment:
You can create getIndexFileCount with tablename and segment and call existing getIndexFileCount method, with extension as null for INDEX_FILE_EXT and refactor the testclass